retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofit.java

```diff
@@ -37,17 +43,25 @@
  * </pre>
  */
 public final class ArmeriaRetrofit {
-    private static final String UNUSED_URL = "http://0.0.0.0";
-
     /**
      * Creates a {@link Retrofit.Builder} with {@link ArmeriaCallFactory} using the specified
      * {@link HttpClient} instance.
      */
     public static Retrofit.Builder builder(HttpClient httpClient) {
         return new Retrofit.Builder()
-                .baseUrl(UNUSED_URL)
+                .baseUrl(convertToSerializationFormatRemovedUri(httpClient.uri()))
                 .callFactory(new ArmeriaCallFactory(httpClient));
     }
 
+    @VisibleForTesting
+    static String convertToSerializationFormatRemovedUri(URI uri) {
+        String uriStr = uri.toString();
```

Comment:
You could just call `uri.getScheme()` or `uri.getRawScheme()` and check the returned value is `null`?